src/modules/Builder/features/ActivityItems/ActivityItems.tsx

```diff
@@ -74,20 +67,22 @@ export const ActivityItems = () => {
     for (const itemIndex of indexListToTrigger) {
       trigger(`${itemsName}.${itemIndex}`);
     }
+
     firstSystemIndex === -1 ? appendItem(item) : insertItem(firstSystemIndex, item);
-    setActiveItemIndex(firstSystemIndex === -1 ? items?.length : firstSystemIndex);
+
+    navigateToItem(item);
   };
 
   const handleInsertItem = (index: number, item?: ItemFormValues) => {
-    const itemToInsert = item ?? getNewActivityItem();
+    const itemToInsert = (item ?? getNewActivityItem()) as ItemFormValues;
```

Comment:
it's better to add the type for returned expression in `getNewActivityItem`

Reply:
done